src/OpenAL/Silk.NET.OpenAL/ALC/ALContext.cs

```diff
@@ -110,6 +111,12 @@ public static unsafe ALContext GetApi(bool soft = false)
             return ret;
         }
 
+        public bool TryGetExtension<T>
+            (out T ext) where T : NativeExtension<ALContext> => 
+            !((ext = IsExtensionPresent(ExtensionAttribute.GetExtensionAttribute(typeof(T)).Name)
+                ? (T) Activator.CreateInstance(typeof(T), Context)
+                : null) is null);
+
```

Comment:
This method should have a `Device*` parameter which you feed in as the first parameter of `IsExtensionPresent`. It's awful we weren't using that in `GetExtension` - we really should have been!

Reply:
Should XML documentation contain a remark `This function doesn't check that the extension is enabled - you will get an error later on if you attempt to call an extension function from an extension that isn't loaded.`? If I understand correctly, `ALContext.TryGetExtension` also doesn't check if it is enabled.

If yes, I think I should update the `AL.TryGetExtension`  XML too.

Reply:
OpenAL doesn't have a concept of enabling extensions like Vulkan does - with Vulkan if you try to get the address to a function in an extension which is not enabled, you will get a symbol loading exception whereas OpenAL allows it and automatically grants access to all extensions available on that implementation. There is `alEnable` which enables the extension's functionality in some cases, but that's different to enabling the extension itself. So I would say leave that out.